scripts/cluster-upgrade.sh

```diff
@@ -20,12 +34,140 @@ then
     unset OCP_VERSION_TO
 fi
 
-UPGRADE_STARTED=""
-UPGRADE_PROGRESSING=""
-UPGRADE_NOT_POSSIBLE=""
-UPGRADE_DONE=""
+log() {
+    CD_NAME=$1
+    STAGE=$2
+    MESSAGE=$3
+    
+    echo "$(date "+%Y-%m-%d_%H.%M.%S") - $CD_NAME - $STAGE - $MESSAGE"
+}
+
+setup() {
+    CD_NAMESPACE=$1
+    CD_NAME=$2
+
+    # get kubeconfig so we can check status of cluster's nodes (extra capacity)
+    oc -n $CD_NAMESPACE get secrets "${CD_NAME}-admin-kubeconfig" -o jsonpath='{.data.kubeconfig}' | base64 -d > $TMP_DIR/kubeconfig-$CD_NAME
```

Comment:
```suggestion
    oc -n $CD_NAMESPACE extract secret/"${CD_NAME}-admin-kubeconfig" --keys=kubeconfig --to=-
```

Reply:
Doesn't really change anything but avoids the need to use base64.  Just a suggestion.